prov/efa/test/efa_unit_test_rnr.c

```diff
@@ -0,0 +1,60 @@
+#include "efa_unit_tests.h"
+#include "rxr_pkt_cmd.h"
+
+/**
+ * @brief this test validate that during RNR queuing and resending,
+ * the "rnr_queued_pkt_cnt" in endpoint and peer were properly updated,
+ * so is the RXR_PKT_ENTRY_RNR_RETRANSMIT flag.
+ */
+void test_efa_rnr_queue_and_resend(struct efa_resource **state)
+{
+	struct efa_resource *resource = *state;
+	struct efa_unit_test_buff send_buff;
+	struct efa_ep_addr raw_addr;
+	struct efa_rdm_ep *efa_rdm_ep;
+	struct efa_rdm_ope *txe;
+	struct rxr_pkt_entry *pkt_entry;
+	size_t raw_addr_len = sizeof(raw_addr);
+	fi_addr_t peer_addr;
+	int ret;
+
+	efa_unit_test_resource_construct(resource, FI_EP_RDM);
+	efa_unit_test_buff_construct(&send_buff, resource, 4096 /* buff_size */);
+	/* Create and register a fake peer */
+	ret = fi_getname(&resource->ep->fid, &raw_addr, &raw_addr_len);
+	assert_int_equal(ret, 0);
+	raw_addr.qpn = 0;
+	raw_addr.qkey = 0x1234;
+
+	ret = fi_av_insert(resource->av, &raw_addr, 1, &peer_addr, 0, NULL);
+	assert_int_equal(ret, 1);
+
+	efa_rdm_ep = container_of(resource->ep, struct efa_rdm_ep, base_ep.util_ep.ep_fid);
+	efa_rdm_ep->base_ep.qp->ibv_qp->context->ops.post_send = &efa_mock_ibv_post_send_save_send_wr;
+	assert_true(dlist_empty(&efa_rdm_ep->txe_list));
+
+	efa_rdm_ep->use_shm_for_tx = false;
+	ret = fi_send(resource->ep, send_buff.buff, send_buff.size, fi_mr_desc(send_buff.mr), peer_addr, NULL /* context */);
```

Comment:
will the send fail for the fake peer?

Reply:
it should not fail, because the `ibv_post_send` has been mocked.

Reply:
I see, didn't realize that. Thx!

Reply:
I also asserted that the ret must be 0.